accounts-db/src/accounts_db/geyser_plugin_utils.rs

```diff
@@ -39,22 +39,25 @@ impl AccountsDb {
     /// in the reverse order of the slots so that an account is only streamed once. At a slot, if the accounts is updated
     /// multiple times only the last write (with highest write_version) is notified.
     pub fn notify_account_restore_from_snapshot(&self) {
-        if self.accounts_update_notifier.is_none() {
+        let Some(accounts_update_notifier) = &self.accounts_update_notifier else {
             return;
-        }
-
-        let mut slots = self.storage.all_slots();
-        let mut notified_accounts: HashSet<Pubkey> = HashSet::default();
-        let mut notify_stats = GeyserPluginNotifyAtSnapshotRestoreStats::default();
+        };
+
+        if accounts_update_notifier.notify_snapshot_disabled() {
+            accounts_update_notifier.notify_end_of_restore_from_snapshot();
```

Comment:
What if we `notify_stats.report();` even in the case when disabled? It should report 0s I guess but it is fine. My assumption is that this report is almost zero cost. 
If so, we could reogranize the code like the following:
```rust
if !accounts_update_notifier.notify_snapshot_disabled() {
         let mut slots = self.storage.all_slots();
            let mut notified_accounts: HashSet<Pubkey> = HashSet::default();
            let mut notify_stats = GeyserPluginNotifyAtSnapshotRestoreStats::default();

            slots.sort_by(|a, b| b.cmp(a));
            for slot in slots {
                self.notify_accounts_in_slot(slot, &mut notified_accounts, &mut notify_stats);
            }
}
accounts_update_notifier.notify_end_of_restore_from_snapshot();
notify_stats.report();
```